src/auth/oauth2client.ts

```diff
@@ -936,4 +932,16 @@ export class OAuth2Client extends AuthClient {
     const buffer = new Buffer(b64String, 'base64');
     return buffer.toString('utf8');
   }
+
+  /**
+   * Returns true iff a token is expired or will expire within
+   * refreshTokenEarlyMillis milliseconds.
+   * If there is no expiry time, assumes the token is not expired or expiring.
+   */
+  protected isTokenExpiring(): boolean {
```

Comment:
So here's a funny question.  Does `refreshTokenEarlyMillis` really need to be configurable?  I had just submitted #248, which sorta blindly adds a 1 second buffer.  Why do we need to make this configurable?  cc @ofrobots @jonparrott

Reply:
1 second may not be enough. For example Cloud Profiler may keep a request open for 1 hour (!). The profiler agent library wants to use 1hr as the timeout to make sure that a token expiring isn't likely going to fail.

Reply:
Fair point :)   I do think we should have a reasonable default here, to cut off the issues in #62.